src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -294,13 +296,35 @@ public void CleanCache()
 
                         token.ThrowIfCancellationRequested();
 
+                        var retries = 0;
+                        var currentExpectedNominations = 0;
+
                         // Drains the queue
                         while (!_pendingRequests.Value.IsCompleted
                             && !token.IsCancellationRequested)
                         {
                             SolutionRestoreRequest discard;
                             if (!_pendingRequests.Value.TryTake(out discard, IdleTimeoutMs, token))
                             {
+                                // check if there are pending nominations
+                                currentExpectedNominations = _solutionManager.Value.GetExpectedProjectsNomination();
```

Comment:
thought so, but then there still could be cases where new projects are being created which are still missing nominations when it tries to run an auto restore which is why I didn't cache it. And also figuring out if projects are nominated or not, is a quick check so doesn't impact perf as well.